src/iidm/Connectable.cpp

```diff
@@ -34,6 +35,21 @@ Terminal& Connectable::addTerminal(std::unique_ptr<Terminal>&& terminal) {
     return *m_terminals.back();
 }
 
+void Connectable::attachTerminal(Terminal& oldTerminal, const std::string& /*oldConnectionInfo*/, VoltageLevel& voltageLevel, std::unique_ptr<Terminal>&& terminal) {
+    // first, attach new terminal to connectable and to voltage level of destination, to ensure that the new terminal is valid
+    terminal->setConnectable(stdcxx::ref(*this));
+    voltageLevel.attach(*terminal, false);
+
+    // then we can detach the old terminal, as we now know that the new terminal is valid
+    oldTerminal.getVoltageLevel().detach(oldTerminal);
+
+    // replace the old terminal by the new terminal in the connectable
+    auto it = std::find_if(m_terminals.begin(), m_terminals.end(), [&oldTerminal](const std::unique_ptr<Terminal>& term) {
+        return stdcxx::areSame(*term, oldTerminal);
+    });
+    *it = std::move(terminal);
```

Comment:
Is it working properly if the terminal is not found (could it really happen)?

Reply:
Yes, it would crash... But I don't see how it could be possible, because the `Connectable` has a list of `Terminal`, and `oldTerminal` is one of them (by definition).

Maybe we should throw an `AssertionError` would be more explicit?

Reply:
TODO: write a unit test with a detached connectable